### PR TITLE
 Update JobQuotaExceededException to include the associated http error reason

### PR DESCRIPTION
**Pull request by abhipsaMisra**
These tests have been failing consistently in our nightly, with the job quota having been exceeded. Marking them to run serially.

### CODE REVIEW

File: iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
 {
-    using System;
-
+    /// <summary>
+    /// Exception thrown when IoT Hub exceeds the available quota for active jobs
+    /// </summary>
     [Serializable]
     public sealed class JobQuotaExceededException : IotHubException
     {
+        private const string JOB_QUOTA_EXCEEDED_MESSAGE = "Job quota has been exceeded";
```
Comment (drwill-ms) [inline iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs:14]: lol why screaming snake case?

File: iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
         public JobQuotaExceededException()
-            : base("Job quota has been exceeded")
+            : base(JOB_QUOTA_EXCEEDED_MESSAGE)
         {
+        }
 
+        /// <summary>
+        /// Initializes an instance of JobQuotaExceededException with the message returned in the Http response
+        /// </summary>
+        /// <param name="message"></param>
```
Comment (drwill-ms) [inline iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs:27]: fill in

File: iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/service/src/Common/Exceptions/JobQuotaExceededException.cs:14]: oops, have been looking at java code a bit too much; will change!

**Comment by abhipsaMisra** [comment]
Keeping this change on hold for now; service is currently swallowing the exception message details (which contains the currently active job IDs). Once service passes that information to the client, we can update the SDK to forward the information to the user.

**Pull request merged**